spyder/api/shellconnect/status.py

```diff
@@ -1,56 +1,126 @@
+        -------
+        None
+        """
         super().__init__(parent)
 
-        # Mapping from shellwidgets to their corresponding statuses
-        self.shellwidget_to_status = {}
+        self.shellwidget_to_status: dict[
+            spyder.plugins.ipyconsole.widgets.shell.ShellWidget, Any
+        ] = {}
+        """
+        Mapping from shell widgets to their corresponding statuses.
+
+        The specific type of status information stored will depend on
+        what the implementing class needs to display.
+        """
 
-        # Current associated shellwidget
-        self.current_shellwidget = None
+        self.current_shellwidget: (
+            spyder.plugins.ipyconsole.widgets.shell.ShellWidget | None
+        ) = None
+        """Currently active associated shell widget."""
 
     # ---- Public API
     # -------------------------------------------------------------------------
-    def update_status(self, status):
+    def update_status(self, status: Any):
         """
         Actions to take to update the widget status after switching consoles.
+
+        Parameters
+        ----------
+        status : Any
+            The shell widget status information needed by the status widget.
+            To be defined by the implementing class.
         """
         raise NotImplementedError
 
-    def on_kernel_start(self, shellwidget):
-        """Actions to take when the kernel starts."""
+    def on_kernel_start(
+        self, shellwidget: spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+    ) -> None:
+        """Actions to take when the kernel starts.
+
+        Parameters
+        ----------
+        shellwidget : spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+            DESCRIPTION.
+
+        Returns
+        -------
+        None
+        """
         raise NotImplementedError
 
-    def config_spyder_kernel(self, shellwidget):
-        """Actions to take to configure the kernel."""
+    def config_spyder_kernel(
+        self, shellwidget: spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+    ) -> None:
+        """Actions to take to configure the kernel.
+
+        Parameters
+        ----------
+        shellwidget : spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+            DESCRIPTION.
+
+        Returns
+        -------
+        None
+        """
         pass
 
     # ---- ShellConnectMixin API
     # -------------------------------------------------------------------------
-    def set_shellwidget(self, shellwidget):
+    def set_shellwidget(
+        self, shellwidget: spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+    ) -> None:
         """
-        Actions to take when setting (i.e. giving focus to) a shellwidget.
+        Actions to take when setting (i.e. giving focus to) a shell widget.
+
+        Parameters
+        ----------
+        shellwidget : spyder.plugins.ipyconsole.widgets.shell.ShellWidget
+            The kernel shell widget to associate with the current widget.
```

Comment:
```suggestion
            The shell widget for which actions will be taken.
```

Reply:
Since the "kernel shell widget" find-replace change outdated this comment, I applied it as part of the rebase in modified form to be clearer, more correct and more specific to what the function is actually doing per its summary line and body code:

```suggestion
            The shell widget that will be given focus.
```